src/core/Element.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <QDebug>
+#include <QString>
+
+class Element
+{
+  public:
+    Element() = default;
+    explicit Element(QString descr, int addr);
+    Element(const Element &) = default;
+    Element(Element &&)      = default;
+    virtual ~Element()       = default;
+
+    Element &operator=(const Element &) = default;
+    Element &operator=(Element &&) = default;
+
+    bool operator==(const Element &other) const;
+    bool operator!=(const Element &other) const;
+
+    QString description() const noexcept;
+    virtual int address() const noexcept;
+
+    virtual void setValue(int8_t val)
+    {
+        (void)(val);
+    }
+    virtual void setValue(int16_t val)
+    {
+        (void)(val);
+    }
+
+    virtual int value() const
+    {
+        return 0;
+    }
```

Comment:
Since the largest value inherited is an _int16_t_ whit this _int_ you are wasting 2 bytes.
I suggest changing `virtual int value() const` to `virtual int16_t value() const`